tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example_raw_scalars/static/views.js

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+
+// Generic view builders.
+
+/**
+ * @param {!Map<string, !Array<Object>>} tagsToScalars
+ * @return {!DocumentFragment}
+ */
+export function createPreviews(tagsToScalars) {
+  const fragment = document.createDocumentFragment();
+
+  if (!tagsToScalars.size) {
+    const messageElement = createElement('h2');
+    messageElement.textContent = 'No tags found.';
+    fragment.appendChild(messageElement);
+    return fragment;
+  }
+
+  /**
+   * For each tag, build UI in this form:
+   *   <div class="preview">
+   *     <div class="tagname">${tag}</div>
+   *     <textarea class="preview-text">${result}</textarea>
+   *   </div>
+   */
+  for (let [tag, scalars] of tagsToScalars) {
+    const previewEl = createElement('div', 'preview');
+    const tagNameEl = createElement('div', 'tagname');
+    const textPreviewEl = createElement('textarea', 'preview-text');
+    tagNameEl.textContent = tag;
+    textPreviewEl.textContent =
+        scalars ? JSON.stringify(scalars) : 'No scalar data found.';
+    previewEl.appendChild(tagNameEl);
+    previewEl.appendChild(textPreviewEl);
+    fragment.appendChild(previewEl);
+  }
+  return fragment;
+}
+
+/**
+ * @param {!Array<string>} runs
+ * @return {!HTMLSelectElement}
+ */
+export function createRunSelector(runs) {
+  /**
+   * Build a component in this form:
+   *   <select class='run-selector>
+   *     <option value='${run}'>${run}</option>
+   *     ...
+   *   </select>
+   */
+  const element = createElement('select', 'run-selector');
+  if (!runs) {
```

Comment:
What’s the purpose of this? If `runs` is a non-nullable `Array<string>`,
then `!runs` will always be `false`. If `if (!runs.length)` is what was
meant, can’t we just drop the condition entirely, as the `for`-loop will
be a no-op? (At which point the comment might be superfluous, too.)

Reply:
You're right, it's not needed.  Removed.